src/reactor.cc

```diff
@@ -0,0 +1,373 @@
+#include "reactor.h"
+
+#include "boost/shared_ptr.hpp"
+
+namespace tricycle {
+
+Reactor::Reactor(cyclus::Context* ctx) : cyclus::Facility(ctx) {
+  // capacities set somewhat arbitrarily
+  fuel_tracker.Init({&tritium_storage}, 1000.0);
+  blanket_tracker.Init({&blanket}, 100000.0);
+}
+
+std::string Reactor::str() {
+  return Facility::str();
+}
+
+void Reactor::Tick() {
+  if (sufficient_tritium_for_operation) {
+    SequesterTritium();
+    OperateReactor(TBR);
+    blanket_fill_policy.Start();
+    Reactor::RecordStatus("Online", fusion_power);
+  } else {
+    Reactor::RecordStatus("Shut-down", 0);
+  }
+
+  DecayInventory(tritium_storage);
+  DecayInventory(tritium_excess);
+  DecayInventory(tritium_sequestered);
+
+  ExtractHelium(tritium_storage);
+  ExtractHelium(tritium_excess);
+
+  if (!tritium_storage.empty() && sufficient_tritium_for_operation) {
+    double surplus = std::max(
+      tritium_storage.quantity() - reserve_inventory, 0.0);
+
+    if (surplus > 0.0) {
+
+      tritium_excess.Push(tritium_storage.Pop(surplus));
+      CombineInventory(tritium_excess);
+
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from storage to excess");
+    }
+  }
+
+  // This pulls out some of the blanket each timestep so that fresh blanket can
+  // be added.
+  double blanket_turnover = blanket_size * blanket_turnover_rate;
+  if (context()->time() % blanket_turnover_frequency == 0) {
+    if (!blanket.empty() && blanket.quantity() >= blanket_turnover) {
```

Comment:
Since both conditions require `!blanket.empty()` maybe combine that with the first condition to simplify further?

Reply:
Yep. You're totally right. Done.